src/network/protocol/messages/checkorder.zig

```diff
@@ -0,0 +1,108 @@
+const std = @import("std");
+const protocol = @import("../lib.zig");
+const Sha256 = std.crypto.hash.sha2.Sha256;
+
+pub const CheckOrderMessage = struct {
+    version: u32,
+    order_id: [32]u8,
+    item_type: [16]u8,
+    item_amount: u64,
+    payment_type: [16]u8,
+    payment_amount: u64,
+    expiration_time: u64,
+
+    pub fn name() *const [12]u8 {
+        return protocol.CommandNames.CHECKORDER ++ [_]u8{0} ** 3;
+    }
+
+    pub fn serialize(self: *const CheckOrderMessage, allocator: std.mem.Allocator) ![]u8 {
```

Comment:
why are you doing things differently for this one. Keep the same method as usual